src/main/java/com/google/starfish/servlets/UserLogoutServlet.java

```diff
@@ -0,0 +1,44 @@
+package com.google.starfish.servlets;  
+
+import java.io.IOException;  
+import javax.servlet.ServletException;  
+import javax.servlet.annotation.WebServlet;  
+import javax.servlet.http.HttpServlet;  
+import javax.servlet.http.HttpSession;  
+import javax.servlet.http.Cookie;  
+import javax.servlet.http.HttpServletRequest;  
+import javax.servlet.http.HttpServletResponse;  
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/** Servlet that handles user account logout */  
+@WebServlet("/user-logout")  
+public class UserLogoutServlet extends HttpServlet {
+
+  private static final Logger LOGGER = Logger.getLogger(UserLogoutServlet.class.getName());
+  private final String COOKIE_NAME = "SFCookie";
+
+  @Override 
+  public void doPost(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
+    // Cookie is unused for now, but will be important once sessions begin to be stored in DB
+    Cookie[] cookies = req.getCookies();
+    for (Cookie cookie : cookies) {
```

Comment:
I think this loop will iterate over all cookies. An optimization could be to first find the cookie and then modify it's attributes. This way if the cookie is one of the first, other cookies don't need to be inspected.

Reply:
This is a good point! If we find the Cookie we're looking for we should simply break out of this loop. If the Cookies array was already sorted alphabetically we could have used binary search to go from O(n) to O(logn) but since I don't think it comes to us ordered, it wouldn't be efficient to sort first in O(nlogn) and then run binary search, so I don't think we can do better than O(n) for now. Best thing is probably just to break out of the loop at the end of the IF statement.